src/main/scala/org/broadinstitute/dsde/workbench/sam/util/ShadowCaster.scala

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.dsde.workbench.sam.util
+import cats.effect._
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.duration.MILLISECONDS
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ShadowCaster extends LazyLogging {
+  implicit val executionContext: ExecutionContext
+
+  protected def runWithShadow[T](functionName: String, real: IO[T], shadow: IO[T]): IO[T] = {
+
+    val realFuture = measure(real).unsafeToFuture()
+    val shadowFuture = measure(shadow).unsafeToFuture()
+
+    Future.reduceLeft[(T, Long), (T, Long)](List(realFuture, shadowFuture)) { case ((realResult, realTime), (shadowResult, shadowTime)) =>
```

Comment:
i find this hard to read, my brain goes "real shadow result real real time shadow time result". how about a `TimedResult(result: T, time: Long)` case class to pass around fewer parameters

Reply:
I agree that extracting the tuples would make this easier to read